src/litdata/streaming/downloader.py

```diff
@@ -188,10 +198,42 @@ def download_file(self, remote_filepath: str, local_filepath: str) -> None:
 }
 
 
+_DOWNLOADERS = {
+    "s3://": "s3",
+    "gs://": "gs",
+    "azure://": "abfs",
+    "local:": "file",
+    "": "file",
+}
+
+
+class FsspecDownloader(Downloader):
```

Comment:
this class can replace all others. And, we can support many more providers, like, huggingface, dropbox, DVC, and many others.

Reply:
One con that I can see is:

- We might lose on the faster S3 bcoz of not utilizing `s5cmd` (if fsspec doesn't use it internally).

Or we can handle this special case separately.

Reply:
Yes, we can probably use it only if installed.

Reply:
cool, so am I on the right track?

If yes, then I first need to figure out the `storage option` stuff and test on all three providers. And, then ask for another round of review.